service/insights_v3.go

```diff
@@ -0,0 +1,24 @@
+package service
+
+import (
+	. "github.com/talbright/go-desk/types"
+	. "github.com/talbright/go-desk/resource"
+	"net/http"
+)
+
+type InsightsV3Service struct {
+	client *Client
+}
+
+func (s *InsightsV3Service) Report(query *InsightsV3Query) (*InsightsV3Report, *http.Response, error) {
```

Comment:
Question: how does this method handle insightsv3 rate limits as described in the docs (http://dev.desk.com/API/insights3/#meta-show)? Note that the rate limiting doesn't utilize header fields like rate limiting does in the rest of the API. 

> Rate Limits
In addition to the general API rate limits described here, the /v2/insights3/ endpoints have an additional rate limit: due to the long-running nature of these requests and the amount of data that they return, only a single /V2/insights3/ outstanding request is allowed at a time. Any request made while another /v2/insights3/ request is outstanding will receive this body:

```json
{
  "message": "Too Many Requests"
}
```

I think it will handle the error gracefully but just wanted to be sure you checked. 

Perhaps adding an integration test for this would be a good idea...